lsusb.c

```diff
@@ -2200,29 +2200,34 @@ static void dump_ccid_device(const unsigned char *buf)
 		fputs("          Auto clock change\n", stdout);
 	if ((us & 0x0020))
 		fputs("          Auto baud rate change\n", stdout);
-	if ((us & 0x0040))
+	if ((us & (0x0040 | 0x0080)) == 0x0040)
 		fputs("          Auto parameter negotiation made by CCID\n", stdout);
-	else if ((us & 0x0080))
+	else if ((us & (0x0040 | 0x0080)) == 0x0080)
```

Comment:
This check happens on the if line below again, so this isn't correct anymore.

Reply:
Why do you say that? The expression 'us & (0x40 | 0x80)' (is the same as us & 0xc0) could have 4 values: 0x00, 0x40, 0x80 or 0xc0. the first if checks for 0x40, the second for 0x80 and the last one for 0xc0, which is an inconsistent value with conflicting flags, hence the warning.

Reply:
Ah, my fault, yes, you are correct.